src/coord/dist_aggregate.c

```diff
@@ -125,29 +126,8 @@ static void netCursorCallback(MRIteratorCallbackCtx *ctx, MRReply *rep) {
   }
 
   // Push the reply down the chain
-  if (isResp3) // RESP3
-  {
-    MRReply *map = MRReply_ArrayElement(rep, 0);
-    MRReply *results = NULL;
-    if (map && MRReply_Type(map) == MR_REPLY_MAP) {
-      results = MRReply_MapElement(map, "results");
-      if (cmd->forProfiling) results = MRReply_MapElement(results, "results"); // profile has an extra level
-      if (results && MRReply_Type(results) == MR_REPLY_ARRAY && MRReply_Length(results) > 0) {
-        MRIteratorCallback_AddReply(ctx, rep); // to be picked up by getNextReply
-        // User code now owns the reply, so we can't free it here ourselves!
-        rep = NULL;
-      }
-    }
-  }
-  else // RESP2
-  {
-    MRReply *results = MRReply_ArrayElement(rep, 0);
-    if (results && MRReply_Type(results) == MR_REPLY_ARRAY && MRReply_Length(results) > 1) {
-      MRIteratorCallback_AddReply(ctx, rep); // to be picked up by getNextReply
-      // User code now owns the reply, so we can't free it here ourselves!
-      rep = NULL;
-    }
-  }
+  MRIteratorCallback_AddReply(ctx, rep); // to be picked up by getNextReply
+  rep = NULL; // User code now owns the reply, so we can't free it here ourselves!
```

Comment:
Essential validation is done previously. We need to pass the reply to the thread so we don't lose the profile info, even if there are no results.
This logic was partially done by the reader already, and now only by the reader and after extracting the profile data